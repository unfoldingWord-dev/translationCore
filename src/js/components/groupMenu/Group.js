import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import PropTypes from 'prop-types';
import CircularProgress from 'material-ui/CircularProgress';
import { Glyphicon } from 'react-bootstrap';
import * as Style from './Style';

class Group extends React.Component {

  render() {
    let style = this.props.active ? Style.menuItem.heading.current : Style.menuItem.heading.normal;

    let glyphAction = this.props.active ? this.props.actions.groupMenuExpandSubMenu : this.props.openGroup;
    let expandedGlyph = (
      <Glyphicon glyph="chevron-down" style={{ float: 'right', marginTop: '3px' }} onClick={glyphAction.bind(this, false)} />
    );
    let collapsedGlyph = (
      <Glyphicon glyph="chevron-right" style={{ float: 'right', marginTop: '3px' }} onClick={glyphAction.bind(this, true)} />
    );

    const { isSubMenuExpanded } = this.props.groupMenuReducer;
  
    return (
      <MuiThemeProvider>
        <div>
          <div style={style} >
            {this.props.active && isSubMenuExpanded ? expandedGlyph : collapsedGlyph}
            <div onClick={this.props.openGroup}>
            <div style={{ marginRight: '10px', float: 'left', border: 'white solid 3px', borderRadius: '50%', width: '20px', height: '20px'}}>
              <CircularProgress
                mode="determinate"
                value={this.props.progress * 100}
                thickness={3}
                size={20}
                color={ this.props.progress ? "var(--accent-color-light)" : 'white'}
                style={{right:3, bottom:3}}
              />
              </div>
              {this.props.groupIndex.name}
            </div>
          </div>
          {this.props.active && isSubMenuExpanded ? this.props.getGroupItems(this) : null}
        </div>
      </MuiThemeProvider>
    );
  }

}

Group.propTypes = {
  groupMenuReducer: PropTypes.any.isRequired,
  actions: PropTypes.shape({
    groupMenuExpandSubMenu: PropTypes.func.isRequired
  }),
  openGroup: PropTypes.any.isRequired,
  progress: PropTypes.any.isRequired,
  groupIndex: PropTypes.any.isRequired,
  getGroupItems: PropTypes.func.isRequired,
  active: PropTypes.any.isRequired
};

export default Group;
