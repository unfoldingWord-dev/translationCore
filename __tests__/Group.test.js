/* eslint-disable no-console */
/* eslint-env jest */
import React from 'react';
import Group from '../src/js/components/groupMenu/Group';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Tests for Group', () => {
  test('Test Group', () => {
    const props = {
      groupMenuReducer: {
        isSubMenuExpanded: true
      },
      actions: {
        groupMenuExpandSubMenu: jest.fn()
      },
      openGroup: jest.fn(),
      progress: .5,
      groupIndex: {
        name: 'God'
      },
      getGroupItems: jest.fn(),
      active: true
    };

    // when
    const wrapper = shallow(
      <Group {...props} />
    );

    // then
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
