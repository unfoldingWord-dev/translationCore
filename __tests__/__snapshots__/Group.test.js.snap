// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Group Test Group 1`] = `
<MuiThemeProvider>
  <div
    className="group"
  >
    <div
      style={
        Object {
          "backgroundColor": "var(--accent-color)",
          "borderBottom": "1px solid var(--background-color)",
          "color": "var(--reverse-color)",
          "cursor": "pointer",
          "display": "block",
          "fontWeight": "bold",
          "paddingBottom": "10px",
          "paddingLeft": "15px",
          "paddingRight": "5px",
          "paddingTop": "7px",
        }
      }
    >
      <Glyphicon
        bsClass="glyphicon"
        glyph="chevron-down"
        onClick={[Function]}
        style={
          Object {
            "float": "right",
            "marginTop": "3px",
          }
        }
      />
      <div
        onClick={[Function]}
      >
        <div
          style={
            Object {
              "border": "white solid 3px",
              "borderRadius": "50%",
              "float": "left",
              "height": "20px",
              "marginRight": "10px",
              "width": "20px",
            }
          }
        >
          <CircularProgress
            color="var(--accent-color-light)"
            max={100}
            min={0}
            mode="determinate"
            size={20}
            style={
              Object {
                "bottom": 3,
                "right": 3,
              }
            }
            thickness={3}
            value={50}
          />
        </div>
        God
      </div>
    </div>
  </div>
</MuiThemeProvider>
`;
